src/extractor/edge_based_graph_factory.cpp

```diff
@@ -442,9 +445,14 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
 
                 distance += turn_penalty;
 
-                BOOST_ASSERT(m_compressed_edge_container.HasEntryForID(edge_from_u));
+                const NodeID smaller = node_u < node_v ? node_u : node_v;
+                const NodeID larger = node_u > node_v ? node_u : node_v;
+                const EdgeID smaller_to_larger_edge = m_node_based_graph->FindEdge(smaller, larger);
+                const EdgeData &smaller_to_larger_edge_data = m_node_based_graph->GetEdgeData(smaller_to_larger_edge);
+                const bool is_encoded_forwards = smaller_to_larger_edge_data.edge_id == SPECIAL_NODEID ? node_u > node_v : node_u < node_v;
```

Comment:
This logic is my least favorite part of the PR, but was the trick that ultimately fixed the errors on the branch. Edges are "zipped" together in the `InsertEdgeBasedNode` method, determining which direction is "forwards" and "backwards" by the order of parameters passed to `CompressedEdge::ZipEdges`. The [calling logic in `InsertEdgeBasedNode`](https://github.com/Project-OSRM/osrm-backend/blob/master/src/extractor/edge_based_graph_factory.cpp#L280-L283) is set up so that the edge will always be oriented from the "smaller" node id to the "larger" node id unless the edge from the smaller to larger node is non-existent, i.e. the edge_id is `SPECIAL_NODEID`. I've seen the non-existent edge happen when the edge from u to v is a reversed oneway.

@MoKob @daniel-j-h do you have any good ideas for going about this in a less-brittle way? As mentioned [here](https://github.com/Project-OSRM/osrm-backend/pull/2717/files#r79394824), we probably want to move the zipping to another part of the extraction process than `EdgeBasedGraphFactory::InsertEdgeBasedNode`. But I'm not sure there's another place in the extraction process where we have the forward and reverse edge id's for a pair of node-based-nodes.